File: Python/large_truck.py
from Truck import Truck
from abc import ABC

class LargeTruck(Truck, ABC):
    MAX_VOLUME = 4000
    MAX_WEIGHT = 8000

    def add_pack(self, pack):
        if self.total_volume() + pack.get_volume() <= LargeTruck.MAX_VOLUME and \
                self.total_weight() + pack.get_weight() <= LargeTruck.MAX_WEIGHT:
            self.packages.append(pack)
        else:
            return False

    def total_volume(self):
        total = 0
        for pack in self.packages:
            total += pack.get_volume()
        return total

    def total_weight(self):
        total = 0
        for pack in self.packages:
            total += pack.get_weight()
        return total

File: CPP/LargeTruck.cpp
#include "headers/LargeTruck.h"

LargeTruck::LargeTruck() {
    truckID = count;
}

int LargeTruck::getMaxWeight() const {
    return MAX_WEIGHT;
}

int LargeTruck::getMaxVolume() const {
    return MAX_VOLUME;
}

void LargeTruck::addPack(const Pack &p) {
    if (totalPacksVolume() + p.getVolume() <= MAX_VOLUME &&
        totalPacksWeight() + p.getWeight() <= MAX_WEIGHT) {
        packages.add(p);
    }
}

int LargeTruck::totalPacksVolume() {
    int total = 0;
    for (int i = 0; i < packages.size(); i++) {
        total += packages.get(i).getVolume();
    }
    return total;
}

int LargeTruck::totalPacksWeight() {
    int total = 0;
    for (int i = 0; i < packages.size(); i++) {
        total += packages.get(i).getWeight();
    }
    return total;
}